lib/rules/state-usage.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @fileoverview Prevent unused state or using state that has never been set
+ * @author Thai Pangsakulyanont @ Taskworld.com
+ */
+'use strict';
+
+var componentUtil = require('../util/component');
+var ComponentList = componentUtil.List;
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = function(context) {
```

Comment:
This should use the [new eslint rule format](http://eslint.org/blog/2016/07/eslint-new-rule-format)